File: Chapter-02-Fundamentals-2/Character-Art/Character-Art.js
// Chapter-2: Fundamentals-2
// Character-Art
// Refer "Star-Art"
// Derive from the above and draw any character passed (not just 'star')

// Assume the argument passed is a positive integer <= 75
// Assume the character argument is a single character
function drawLeftChars(num, char){
  var str = ""
  for(let i = 1; i <= num; i++){
    str += char
  }
  console.log(str)
}

function drawRightChars(num, char){
  var str = ""
  for(let i = 1; i <= 75 - num; i++){
    str += "-"
  }
  for(let j = 1; j <= num; j++){
    str += char
  }
  console.log(str)
}

function drawCenterChars(num, char){
  var str = ""
  for(let i = 1; i <= Math.floor((75 - num)/2); i++){
    str += "-"
  }
  for(let j = 1; j <= num; j++){
    str += char
  }
  var len = str.length
  for(let k = 1; k <= 75 - len; k++){
    str += "-"
  }
  console.log(str)
}

drawLeftChars(35, '@')
drawRightChars(40, '#')
drawCenterChars(40, '~')

File: Chapter-02-Fundamentals-2/Character-Art/Character-Art.py
# Chapter-2: Fundamentals-2
# Character-Art
# Refer "Star-Art"
# Derive from the above and draw any character passed (not just 'star')

# Assume the argument passed is a positive integer <= 75
# Assume the character argument is a single character
def drawLeftChars(num, char):
    print("{}").format(char * num)

def drawRightChars(num, char):
    str1 = "-" * (75 - num)
    str2 = char * num
    print("{}").format(str1 + str2)

def drawCenterChars(num, char):
    str1 = "-" * ((75 - num)/2)
    str2 = char * num
    str3 = "-" * (75 - ((75 - num)/2 + num))
    print("{}").format(str1+str2+str3)

drawLeftChars(35, '@')
drawRightChars(40, '#')
drawCenterChars(35, '~')
